### ISSUE TITLE
32-bit and 64-bit type mismatches

### ISSUE THREAD

**Issue by shepmaster** [Osspial/glyphydog#2]
```
error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:226:17
    |
226 |                 face_index,
    |                 ^^^^^^^^^^ expected i64, found i32
help: you can cast an `i32` to `i64`, which will sign-extend the source value
    |
226 |                 face_index.into(),
    |                 ^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:237:75
    |
237 |                         if FT_Error(0) != ft::FT_Load_Sfnt_Table(ft_face, tag, 0, ptr::null_mut(), &mut len) {
    |                                                                           ^^^ expected u64, found u32
help: you can cast an `u32` to `u64`, which will zero-extend the source value
    |
237 |                         if FT_Error(0) != ft::FT_Load_Sfnt_Table(ft_face, tag.into(), 0, ptr::null_mut(), &mut len) {
    |                                                                           ^^^^^^^^^^

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:242:75
    |
242 |                         if FT_Error(0) != ft::FT_Load_Sfnt_Table(ft_face, tag, 0, buf.as_mut_ptr() as *mut ft::FT_Byte, &mut len) {
    |                                                                           ^^^ expected u64, found u32
help: you can cast an `u32` to `u64`, which will zero-extend the source value
    |
242 |                         if FT_Error(0) != ft::FT_Load_Sfnt_Table(ft_face, tag.into(), 0, buf.as_mut_ptr() as *mut ft::FT_Byte, &mut len) {
    |                                                                           ^^^^^^^^^^

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:247:47
    |
247 |                             buf.as_mut_ptr(), len, HB_MEMORY_MODE_WRITABLE,
    |                                               ^^^ expected u32, found u64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:296:17
    |
296 |                 font_buffer.len() as c_int,
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^ expected i64, found i32
help: you can cast an `i32` to `i64`, which will sign-extend the source value
    |
296 |                 (font_buffer.len() as c_int).into(),
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:297:17
    |
297 |                 face_index,
    |                 ^^^^^^^^^^ expected i64, found i32
help: you can cast an `i32` to `i64`, which will sign-extend the source value
    |
297 |                 face_index.into(),
    |                 ^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:370:53
    |
370 |         unsafe{ ft::FT_Get_Char_Index(self.ft_face, c as u32) }
    |                                                     ^^^^^^^^ expected u64, found u32
help: you can cast an `u32` to `u64`, which will zero-extend the source value
    |
370 |         unsafe{ ft::FT_Get_Char_Index(self.ft_face, (c as u32).into()) }
    |                                                     ^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:384:35
    |
384 |                 FT_Error(0) => Ok(advance),
    |                                   ^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:413:22
    |
413 |             x_scale: size_metrics.x_scale,
    |                      ^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:414:22
    |
414 |             y_scale: size_metrics.y_scale,
    |                      ^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:415:23
    |
415 |             ascender: size_metrics.ascender,
    |                       ^^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:416:24
    |
416 |             descender: size_metrics.descender,
    |                        ^^^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:417:21
    |
417 |             height: size_metrics.height,
    |                     ^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:418:26
    |
418 |             max_advance: size_metrics.max_advance,
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:438:24
    |
438 |                 width: face_size.width as i32,
    |                        ^^^^^^^^^^^^^^^^^^^^^^ expected i64, found i32
help: you can cast an `i32` to `i64`, which will sign-extend the source value
    |
438 |                 width: (face_size.width as i32).into(),
    |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:439:25
    |
439 |                 height: face_size.height as i32,
    |                         ^^^^^^^^^^^^^^^^^^^^^^^ expected i64, found i32
help: you can cast an `i32` to `i64`, which will sign-extend the source value
    |
439 |                 height: (face_size.height as i32).into(),
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:590:33
    |
590 |             dims: DimsBox::new2(ft_metrics.width, ft_metrics.height),
    |                                 ^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:590:51
    |
590 |             dims: DimsBox::new2(ft_metrics.width, ft_metrics.height),
    |                                                   ^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:591:40
    |
591 |             hori_bearing: Vector2::new(ft_metrics.horiBearingX, ft_metrics.horiBearingY),
    |                                        ^^^^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:591:65
    |
591 |             hori_bearing: Vector2::new(ft_metrics.horiBearingX, ft_metrics.horiBearingY),
    |                                                                 ^^^^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:592:27
    |
592 |             hori_advance: ft_metrics.horiAdvance,
    |                           ^^^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:593:40
    |
593 |             vert_bearing: Vector2::new(ft_metrics.vertBearingX, ft_metrics.vertBearingY),
    |                                        ^^^^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:593:65
    |
593 |             vert_bearing: Vector2::new(ft_metrics.vertBearingX, ft_metrics.vertBearingY),
    |                                                                 ^^^^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:594:27
    |
594 |             vert_advance: ft_metrics.vertAdvance,
    |                           ^^^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/lib.rs:676:32
    |
676 |         Face::new(buf, unsafe{ (*self.ft_face).face_index }, &self._lib).unwrap()
    |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
  --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/hb_funcs.rs:71:27
   |
71 |     hb_metrics.ascender = ft_metrics.ascender;
   |                           ^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
  --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/hb_funcs.rs:72:28
   |
72 |     hb_metrics.descender = ft_metrics.descender;
   |                            ^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
  --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/hb_funcs.rs:73:27
   |
73 |     hb_metrics.line_gap = ft_metrics.height - (ft_metrics.ascender - ft_metrics.descender);
   |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
  --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/hb_funcs.rs:93:57
   |
93 |     let mut char_index = FT_Get_Char_Index(ffd.ft_face, unicode);
   |                                                         ^^^^^^^ expected u64, found u32
help: you can cast an `u32` to `u64`, which will zero-extend the source value
   |
93 |     let mut char_index = FT_Get_Char_Index(ffd.ft_face, unicode.into());
   |                                                         ^^^^^^^^^^^^^^

error[E0308]: mismatched types
  --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/hb_funcs.rs:96:53
   |
96 |         char_index = FT_Get_Char_Index(ffd.ft_face, 0xF000 + unicode);
   |                                                     ^^^^^^^^^^^^^^^^ expected u64, found u32
help: you can cast an `u32` to `u64`, which will zero-extend the source value
   |
96 |         char_index = FT_Get_Char_Index(ffd.ft_face, (0xF000 + unicode).into());
   |                                                     ^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/hb_funcs.rs:116:63
    |
116 |     let char_index = FT_Face_GetCharVariantIndex(ffd.ft_face, unicode, variation_selector);
    |                                                               ^^^^^^^ expected u64, found u32
help: you can cast an `u32` to `u64`, which will zero-extend the source value
    |
116 |     let char_index = FT_Face_GetCharVariantIndex(ffd.ft_face, unicode.into(), variation_selector);
    |                                                               ^^^^^^^^^^^^^^

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/hb_funcs.rs:116:72
    |
116 |     let char_index = FT_Face_GetCharVariantIndex(ffd.ft_face, unicode, variation_selector);
    |                                                                        ^^^^^^^^^^^^^^^^^^ expected u64, found u32
help: you can cast an `u32` to `u64`, which will zero-extend the source value
    |
116 |     let char_index = FT_Face_GetCharVariantIndex(ffd.ft_face, unicode, variation_selector.into());
    |                                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/hb_funcs.rs:143:13
    |
132 | ) -> hb_position_t
    |      ------------- expected `i32` because of return type
...
143 |             (advance + (1<<9)) >> 10
    |             ^^^^^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/hb_funcs.rs:164:13
    |
154 | ) -> hb_position_t {
    |      ------------- expected `i32` because of return type
...
164 |             (-advance + (1<<9)) >> 10
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/hb_funcs.rs:184:18
    |
184 |             *x = glyph_metrics.horiBearingX - glyph_metrics.vertBearingX;
    |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/hb_funcs.rs:185:18
    |
185 |             *y = glyph_metrics.horiBearingY - (-glyph_metrics.vertBearingY);
    |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: match arms have incompatible types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/hb_funcs.rs:215:5
    |
215 | /     match FT_Get_Kerning(ffd.ft_face, left_glyph, right_glyph, mode as c_uint, &mut kerningv) {
216 | |         FT_Error(0) => kerningv.x,
    | |                        ---------- match arm with an incompatible type
217 | |         _ => 0
218 | |     }
    | |_____^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/hb_funcs.rs:235:33
    |
235 |             extents.x_bearing = glyph_metrics.horiBearingX;
    |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/hb_funcs.rs:236:33
    |
236 |             extents.y_bearing = glyph_metrics.horiBearingY;
    |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/hb_funcs.rs:237:29
    |
237 |             extents.width = glyph_metrics.width;
    |                             ^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/hb_funcs.rs:238:30
    |
238 |             extents.height = glyph_metrics.height;
    |                              ^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/hb_funcs.rs:279:10
    |
279 |     *x = (*glyph.outline.points.offset(point_index as isize)).x;
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/29f3051/src/hb_funcs.rs:280:10
    |
280 |     *y = (*glyph.outline.points.offset(point_index as isize)).y;
    |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

```

**Comment by shepmaster** [comment]
After recent update:

```
error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/fa1432c/src/lib.rs:226:17
    |
226 |                 face_index as i32,
    |                 ^^^^^^^^^^^^^^^^^ expected i64, found i32
help: you can cast an `i32` to `i64`, which will sign-extend the source value
    |
226 |                 (face_index as i32).into(),
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/fa1432c/src/lib.rs:247:47
    |
247 |                             buf.as_mut_ptr(), len as FT_ULong, HB_MEMORY_MODE_WRITABLE,
    |                                               ^^^^^^^^^^^^^^^ expected u32, found u64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/fa1432c/src/lib.rs:676:32
    |
676 |         Face::new(buf, unsafe{ (*self.ft_face).face_index as FT_Long }, &self._lib).unwrap()
    |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected i32, found i64

error[E0308]: mismatched types
   --> /Users/shep/.cargo/git/checkouts/glyphydog-3af71cac8a3449cb/fa1432c/src/hb_funcs.rs:116:84
    |
116 |     let char_index = FT_Face_GetCharVariantIndex(ffd.ft_face, unicode as FT_ULong, variation_selector);
    |                                                                                    ^^^^^^^^^^^^^^^^^^ expected u64, found u32
help: you can cast an `u32` to `u64`, which will zero-extend the source value
    |
116 |     let char_index = FT_Face_GetCharVariantIndex(ffd.ft_face, unicode as FT_ULong, variation_selector.into());
    |                                                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^

```

**Comment by Osspial** [comment]
Fixed by fa1432c17629694717717cab6e30b1dae1d4643d and 0db0cf0eb13829a6e87be09b07b8ddd6fcdc22cb
